### PR TITLE
Fixed GCC Warnings

### PR DESCRIPTION
**Pull request by CraftingInC**
I fixed a lot of the warnings that GCC / MinGWx64 has with raylib. I kept original code where needed so as to work with other compilers by adding an if define.

### CODE REVIEW

File: src/external/rl_gputex.h [hunk 1]
```diff
@@ -771,7 +771,11 @@ void *rl_load_astc_from_memory(const unsigned char *file_data, unsigned int file
 // Module Internal Functions Definition
 //----------------------------------------------------------------------------------
 // Get pixel data size in bytes for certain pixel format
+#ifdef __GNUC__
+static inline int get_pixel_data_size(int width, int height, int format)
```
Comment (raysan5) [inline src/external/rl_gputex.h:775]: Why this function needs to be `inlined`?

File: src/external/stb_truetype.h [hunk 2]
```diff
@@ -500,7 +500,11 @@ int main(int arg, char **argv)
 #define __STB_INCLUDE_STB_TRUETYPE_H__
 
 #ifdef STBTT_STATIC
+#ifdef __GNUC__
+#define STBTT_DEF static inline
```
Comment (raysan5) [inline src/external/stb_truetype.h:504]: Why?

In any case, external libraries shouldn't be modified in raylib but in the original repositories.

**Comment by raysan5** [comment]
@CraftingInC Thanks for the review. I added some comments but I noticed most of the changes are trailing-spaces removal, formatting changes should be moved into a separate PR.

File: src/external/stb_truetype.h (hunk 2 above)
Comment (CraftingInC) [reply src/external/stb_truetype.h:504]: GCC / MinGWx64 would give a warning without the inline being there. Here is more info about INLINE on the official GCC website.

https://gcc.gnu.org/onlinedocs/gcc/Inline.html

**Comment by CraftingInC** [comment]
> 

I saw no trailing space issues on my end. So that must be from something else. Odd that you see that, and I did not.

File: src/external/rl_gputex.h (hunk 1 above)
Comment (CraftingInC) [reply src/external/rl_gputex.h:775]: Check my below answer. To remove the GCC / MinGWx64 WARNING, I posted a link below.

File: src/external/stb_truetype.h (hunk 2 above)
Comment (raysan5) [reply src/external/stb_truetype.h:504]: I'm afraid that link does not answer my question, I know what is an inline function.

**Comment by raysan5** [comment]
@CraftingInC Sorry, I'm not merging this PR, I think it is not needed.

File: src/external/stb_truetype.h (hunk 2 above)
Comment (CraftingInC) [reply src/external/stb_truetype.h:504]: In GCC / MinGWx64 you will get these WARNINGS : defined but not used [-Wunused-function]

To remove those warnings without supressing it, you use the keyword INLINE.

**Pull request closed without merging**